src/Orchard.Web/Modules/Orchard.OpenId/OwinMiddlewares/Google.cs

```diff
@@ -27,7 +29,7 @@ public class Google : IOwinMiddlewareProvider {
             var authenticationOptions = new GoogleOAuth2AuthenticationOptions {
                 ClientId = settings.ClientId,
                 ClientSecret = settings.ClientSecret,
-                CallbackPath = new PathString(settings.CallbackPath)
+                CallbackPath = new PathString(GetCallbackPath(workContext, settings))
```

Comment:
Why to add an extra step of responsibilty on the user?
I prefer less intervention from the user in such scenarios.